packages/snfoundry/contracts/src/YourCollectible.cairo

```diff
@@ -0,0 +1,221 @@
+use starknet::ContractAddress;
+
+#[starknet::interface]
+pub trait IYourCollectible<T> {
+    fn mint_item(ref self: T, recipient: ContractAddress, uri: ByteArray) -> u256;
+}
+
+#[starknet::contract]
+mod YourCollectible {
+    use starknet::storage::Map;
+    use contracts::components::Counter::CounterComponent;
+    use contracts::components::ERC721Enumerable::ERC721EnumerableComponent;
```

Comment:
I expect to get rid of this custum EnumerableComponent and use the provided by [OZ](https://github.com/OpenZeppelin/cairo-contracts/blob/f608acd786af77fec2357860bcbe6c0baf3117fa/packages/token/src/erc721/extensions/erc721_enumerable/erc721_enumerable.cairo#L18)